src/main/java/seedu/contax/model/person/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,28 @@
+package seedu.contax.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public abstract class ContainsKeywordsPredicate implements Predicate<Person> {
+
+    private final List<String> keywords;
+
+    public ContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    public List<String> getKeywords() {
+        return keywords;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof ContainsKeywordsPredicate // instanceof handles nulls
+                && keywords.equals(((ContainsKeywordsPredicate) other).keywords)); // state check
+    }
```

Comment:
I suspect that the equals will say that different subclasses are equal (E.g. EmailContainsKeywordsPredicate will be equal to NameContainsKeywordPredicate) if they have the same keywords, even though they are different types.

Reply:
thanks I change to each subclass override